### DISCUSSION TITLE
I am Facing 404, The route could not be found.

### DISCUSSION THREAD

**Discussion by ArsalanSiddique** [livewire/livewire#5534] [1. Help]
I am Facing 404, The route `public/livewire/message/software-category` could not be found.

I am using Laravel 9 with livewire and my components are loaded perfectly but when I try to update any attribute value of the component, it returns `404 not found`.

here is my **.env**

    APP_URL=https://fyp.devndesigns.com/
    ASSET_URL=https://fyp.devndesigns.com/public

here is my config/livewire.php in which I changed my asset_url but it is not working.

    <?php
    
    return [
    
        /*
        |--------------------------------------------------------------------------
        | Class Namespace
        |--------------------------------------------------------------------------
        |
        | This value sets the root namespace for Livewire component classes in
        | your application. This value affects component auto-discovery and
        | any Livewire file helper commands, like `artisan make:livewire`.
        |
        | After changing this item, run: `php artisan livewire:discover`.
        |
        */
    
        'class_namespace' => 'App\\Http\\Livewire',
    
        /*
        |--------------------------------------------------------------------------
        | View Path
        |--------------------------------------------------------------------------
        |
        | This value sets the path for Livewire component views. This affects
        | file manipulation helper commands like `artisan make:livewire`.
        |
        */
    
        'view_path' => resource_path('views/livewire'),
    
        /*
        |--------------------------------------------------------------------------
        | Layout
        |--------------------------------------------------------------------------
        | The default layout view that will be used when rendering a component via
        | Route::get('/some-endpoint', SomeComponent::class);. In this case the
        | the view returned by SomeComponent will be wrapped in "layouts.app"
        |
        */
    
        'layout' => 'layouts.app',
    
        /*
        |--------------------------------------------------------------------------
        | Livewire Assets URL
        |--------------------------------------------------------------------------
        |
        | This value sets the path to Livewire JavaScript assets, for cases where
        | your app's domain root is not the correct path. By default, Livewire
        | will load its JavaScript assets from the app's "relative root".
        |
        | Examples: "/assets", "myurl.com/app".
        |
        */
    
        'asset_url' => env('ASSET_URL'),
    
        /*
        |--------------------------------------------------------------------------
        | Livewire App URL
        |--------------------------------------------------------------------------
        |
        | This value should be used if livewire assets are served from CDN.
        | Livewire will communicate with an app through this url.
        |
        | Examples: "https://my-app.com", "myurl.com/app".
        |
        */
    
        'app_url' => null,
    
        /*
        |--------------------------------------------------------------------------
        | Livewire Endpoint Middleware Group
        |--------------------------------------------------------------------------
        |
        | This value sets the middleware group that will be applied to the main
        | Livewire "message" endpoint (the endpoint that gets hit everytime
        | a Livewire component updates). It is set to "web" by default.
        |
        */
    
        'middleware_group' => 'web',

**Comment by joshhanley** [comment]
@ArsalanSiddique what patch version of Livewire are you using? Run `composer show -D` in your terminal to find out the exact version. There has been some bugs which has been patch, so if you are using v.2.10.8 I would recommend upgrading to the latest version.

Hope this helps!
